Bugsnag/Helpers/BSGCachesDirectory.m

```diff
@@ -34,4 +35,29 @@ + (NSString*) cachesDirectory {
     return cachesPath;
 }
 
++ (NSString *)getSubdirPath:(NSString *)relativePath {
+    NSString *cachesDir = [self cachesDirectory];
+    NSString *subdirPath = [cachesDir stringByAppendingPathComponent:relativePath];
+    
+    NSFileManager *fileManager = [NSFileManager defaultManager];
+    BOOL isDir;
+    BOOL exists = [fileManager fileExistsAtPath:subdirPath isDirectory:&isDir];
+    if(exists && !isDir) {
+        BSG_KSLOG_ERROR(@"Path %@ exists but is not a directory as expected", subdirPath);
+        // Make the best of it, just return the top-level caches dir.
+        return cachesDir;
+    }
+
+    if(!exists) {
```

Comment:
FYI [Apple's documentation recommends against doing these kind of pre-flight checks](https://developer.apple.com/documentation/foundation/nsfilemanager/1415645-fileexistsatpath#discussion)

> Note
Attempting to predicate behavior based on the current state of the file system or a particular file on the file system is not recommended. Doing so can cause odd behavior or race conditions. It’s far better to attempt an operation (such as loading a file or creating a directory), check for errors, and handle those errors gracefully than it is to try to figure out ahead of time whether the operation will succeed. For more information on file-system race conditions, see Race Conditions and Secure File Operations in Secure Coding Guide.